docs/DeveloperGuide.md

```diff
@@ -276,32 +276,40 @@ With the design considerations, we've chosen the alternatives that provide a bal
 
 ### 4.3 Filter Command
 
-The `filter` command allows user to filter the company list by the application status. The following sequence diagram
-will illustrate the process of performing the `filter` command.
-
-<img src="images/FilterSequenceDiagram.png"/>
+The `filter` command allows user to filter the company list by the application status.
 
 #### 4.3.1 Implementation
+The `filter` command is implemented in the `FilterCommand` class, which uses the `ApplicationStatusPredicate` class. The `ApplicationStatusPredicate` class implements the `Predicate` interface, which allows it to be used in the `Model` interface's `updateFilteredCompanyList(Predicate<Company> predicate)` method.
+
+Given below is an example usage scenario and how the `filter` mechanism behaves at each step.
+
+1. The user enters the input `filter s/PA`.
+2. The `LogicManager` calls `AddressBookParser#parseCommand()` with the user input.
+3. The `AddressBookParser` creates a parser that matches the `filter` command, a `FilterCommandParser` object and uses it to parse the command.
+4. This results in a `FilterCommand` object, which is executed by the `LogicManager`.
+5. The `FilterCommand` object can communicate with the `Model` when it is executed. It calls `Model#filterCompaniesByStatus(Predicate<Company> predicate)` to filter the list of companies by their application status.
+6. The result of the command execution is encapsulated as a `CommandResult` object which is returned back from `Logic`.
 
-The `filter` function is implemented in the `FilterCommand` class and uses the `FilterCommandParser` class to parse the
-arguments. The predicate class implementing `Predicate<Company>` is `ApplicationStatusPredicate`.
-- `ApplicationStatusPredicate` - Predicate to check if the company's application status is the same as the application
-  status specified in the command.
+The following sequence diagram will illustrate the process of performing the `filter` command.
+
+<img src="images/FilterSequenceDiagram.png"/>
+
+<div markdown="block" class="alert alert-info">
+**:information_source: Note:**
+The lifeline for `FilterCommandParser` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+</div>
 
 #### 4.3.2 Design Considerations
 
-**Aspect: UI of the filter command**
+**Aspect: UI of the Filter Command**
 
-- **Alternative 1 (current choice):** The company details panel will be cleared whenever the filter command is executed.
-    - Pros: Users can focus on viewing details of company belonging to the filtered list only, reducing distractions and confusions.
-    - Cons: Users might have to execute the `view` command again to access details of the company that is selected even
-      if that company is in the filtered list, potentially leading to additional steps.
+* **Alternative 1:** The company details panel will still display the details of the company that was selected before the filter command is executed.
```

Comment:
Should be "Company detail panel" instead of "Company details panel"? I tried chatgpting company details panel suggests a panel that contains details about multiple companies, while company detail panel implies a panel that provides detailed info about a single company.

Reply:
Do you think I should put it as 
`CompanyDetailPanel` or just company detail panel will do?

Reply:
Do you think I should do `CompanyDetailPanel` or just company detail panel?